// blocksin-system/dist/Switch.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireWildcard(require("react"));
var SwitchPrimitive = _interopRequireWildcard(require("@radix-ui/react-switch"));
require("./radix-styles/Switch.scss");
var _Flex = _interopRequireDefault(require("./Flex"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
var Switch = /*#__PURE__*/(0, _react.forwardRef)(function (_ref, ref) {
  var id = _ref.id,
    checked = _ref.checked,
    onCheckedChange = _ref.onCheckedChange,
    children = _ref.children,
    disabled = _ref.disabled,
    required = _ref.required;
  return children ? /*#__PURE__*/_react["default"].createElement(_Flex["default"], {
    customClass: "CustomSwitch",
    gap: 200,
    align: "center"
  }, /*#__PURE__*/_react["default"].createElement(SwitchPrimitive.Root, {
    className: "SwitchRoot",
    id: id,
    checked: checked,
    onCheckedChange: onCheckedChange,
    disabled: disabled,
    required: required,
    ref: ref
  }, /*#__PURE__*/_react["default"].createElement(SwitchPrimitive.Thumb, {
    className: "SwitchThumb"
  })), /*#__PURE__*/_react["default"].createElement("label", {
    className: "Label",
    htmlFor: id
  }, children)) : /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(SwitchPrimitive.Root, {
    className: "SwitchRoot",
    id: id,
    checked: checked,
    onCheckedChange: onCheckedChange,
    disabled: disabled,
    ref: ref,
    required: required
  }, /*#__PURE__*/_react["default"].createElement(SwitchPrimitive.Thumb, {
    className: "SwitchThumb"
  })));
});
var _default = exports["default"] = Switch;
