// blocksin-system/dist/UserListItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireDefault(require("react"));
require("./UserListItem.scss");
var _Paragraph = _interopRequireDefault(require("./Paragraph"));
var _Flex = _interopRequireDefault(require("./Flex"));
var _BadgeLevel = _interopRequireDefault(require("./BadgeLevel"));
var _Avatar = _interopRequireDefault(require("./Avatar"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var UserListItem = function UserListItem(_ref) {
  var name = _ref.name,
    badge = _ref.badge,
    avatar = _ref.avatar,
    role = _ref.role,
    children = _ref.children;
  return /*#__PURE__*/_react["default"].createElement(_Flex["default"], {
    direction: "row",
    customClass: "UserListItem",
    justify: "between",
    align: "center",
    gap: 300
  }, /*#__PURE__*/_react["default"].createElement(_Flex["default"], {
    style: {
      gap: "var(--size-150)"
    },
    align: "center"
  }, /*#__PURE__*/_react["default"].createElement(_Avatar["default"], {
    size: "medium",
    avatar: avatar,
    altText: name,
    role: role
  }), /*#__PURE__*/_react["default"].createElement(_Paragraph["default"], {
    size: "medium"
  }, name)), /*#__PURE__*/_react["default"].createElement(_Flex["default"], {
    gap: 100,
    align: "center"
  }, children, /*#__PURE__*/_react["default"].createElement(_Flex["default"], {
    customClass: "BadgeLevelContainer"
  }, /*#__PURE__*/_react["default"].createElement(_BadgeLevel["default"], {
    badge: badge
  }))));
};
var _default = exports["default"] = UserListItem;
